flink-runtime/src/main/java/org/apache/flink/runtime/io/network/NetworkEnvironment.java

```diff
@@ -271,4 +378,112 @@ public boolean isShutdown() {
 			return isShutdown;
 		}
 	}
+
+	// --------------------------------------------------------------------------------------------
+	//  Input and Output Metrics
+	// --------------------------------------------------------------------------------------------
+
+	/**
+	 * Gauge measuring the number of queued input buffers of a task.
+	 */
+	private static final class InputBuffersGauge implements Gauge<Integer> {
```

Comment:
I would move these classes to some `network.metrics` package along with `InputChannelMetrics` and `InputGateMetrics`.